src/raftstore/store/bootstrap.rs

```diff
@@ -87,3 +98,34 @@ pub fn bootstrap_region(engine: &DB,
 
     Ok(region)
 }
+
+
+#[cfg(test)]
+mod tests {
+    use tempdir::TempDir;
+
+    use super::*;
+    use util::rocksdb;
+    use raftstore::store::engine::Peekable;
+    use raftstore::store::keys;
+    use storage::CF_RAFT;
+
+    #[test]
+    fn test_bootstrap() {
+        let path = TempDir::new("var").unwrap();
+        let engine = rocksdb::new_engine(path.path().to_str().unwrap(), &[CF_RAFT]).unwrap();
+
+        assert!(bootstrap_store(&engine, 1, 1).is_ok());
+        assert!(bootstrap_store(&engine, 1, 1).is_err());
+
+        assert!(bootstrap_region(&engine, 1, 1, 1).is_ok());
+        assert!(engine.get_value(&keys::region_state_key(1)).unwrap().is_some());
+        assert!(engine.get_value_cf(CF_RAFT, &keys::raft_state_key(1)).unwrap().is_some());
+        assert!(engine.get_value_cf(CF_RAFT, &keys::apply_state_key(1)).unwrap().is_some());
+
+        assert!(clear_region(&engine, 1).is_ok());
+        assert!(engine.get_value(&keys::region_state_key(1)).unwrap().is_none());
```

Comment:
Should scan instead. Otherwise new key is added in bootstrap_region but not in clear_region, the case can still pass.